### PR TITLE
[Bots][Buildbotsrevamp][1574233]

### PR DESCRIPTION
**Pull request by v-rodriguezd**
Completed reviewing 19 documents under the Build Bots section for structure and language changes. Please review.

### CODE REVIEW

File: msteams-platform/bots/how-to/update-and-delete-bot-messages.md [hunk 1]
```diff
@@ -10,25 +10,27 @@ ms.author: anclear
...
+
 ## Update messages
 
-Your bot can dynamically update messages after sending them. You can use dynamic message updates for scenarios such as poll updates, modifying available actions after a button press, or any other asynchronous state change.
+You can use dynamic message updates for scenarios, such as poll updates, modifying available actions after a button press, or any other asynchronous state change.
 
 The new message need not match the original in type. For example, if the original message contained an attachment, the new message can be a simple text message.
 
-# [C#/.NET](#tab/dotnet)
+# [C# or .NET](#tab/dotnet)
```
Comment (heath-hamilton) [inline msteams-platform/bots/how-to/update-and-delete-bot-messages.md:21]:
```suggestion
# [C#](#tab/dotnet)
```
Adding an "or .NET" makes this inaccurate. You only need C#.

File: msteams-platform/bots/how-to/update-and-delete-bot-messages.md [hunk 2]
```diff
@@ -10,25 +10,27 @@ ms.author: anclear
...
+To update an existing message, pass a new `Activity` object with the existing activity ID to the `UpdateActivityAsync` method of the `TurnContext` class. For more information, see [TurnContextClass](/dotnet/api/microsoft.bot.builder.turncontext?view=botbuilder-dotnet-stable&preserve-view=true).
 
 ```csharp
 var newActivity = MessageFactory.Text("The new text for the activity");
 newActivity.Id = activityId;
 await turnContext.UpdateActivityAsync(newActivity, cancellationToken);
 ```
 
-# [TypeScript/Node.js](#tab/typescript)
+# [TypeScript or Node.js](#tab/typescript)
```
Comment (heath-hamilton) [inline msteams-platform/bots/how-to/update-and-delete-bot-messages.md:31]:
```suggestion
# [TypeScript](#tab/typescript)
```
Don't need "or Node.js".

File: msteams-platform/bots/how-to/update-and-delete-bot-messages.md [hunk 3]
```diff
@@ -50,27 +52,33 @@ update_result = await context.update_activity(new_activity)
...
+* * *
+
+Now that you have updated messages, update the existing card on button selection for incoming activities.
 
 ## Update cards
 
 To update the existing card on button selection, you can use `ReplyToId` of incoming activity.
 
-# [C#/.NET](#tab/dotnet)
+# [C# or .NET](#tab/dotnet)
```
Comment (heath-hamilton) [inline msteams-platform/bots/how-to/update-and-delete-bot-messages.md:81]:
```suggestion
# [C#](#tab/dotnet)
```

File: msteams-platform/bots/how-to/update-and-delete-bot-messages.md [hunk 4]
```diff
@@ -79,7 +87,7 @@ activity.Id = turnContext.Activity.ReplyToId;
 await turnContext.UpdateActivityAsync(activity, cancellationToken);
 ```
 
-# [TypeScript/Node.js](#tab/typescript)
+# [TypeScript or Node.js](#tab/typescript)
```
Comment (heath-hamilton) [inline msteams-platform/bots/how-to/update-and-delete-bot-messages.md:90]:
```suggestion
# [TypeScript](#tab/typescript)
```

File: msteams-platform/bots/how-to/update-and-delete-bot-messages.md [hunk 5]
```diff
@@ -99,14 +107,30 @@ await turn_context.update_activity(updated_activity)
...
+Now that you have updated cards, you can delete messages using the Bot Framework.
+
 ## Delete messages
 
-In the Bot Framework, every message has its own unique activity identifier.
-Messages can be deleted using the Bot Framework's `DeleteActivity` method as shown here.
+In the Bot Framework, every message has its own unique activity identifier. Messages can be deleted using the Bot Framework's `DeleteActivity` method.
 
-# [C#/.NET](#tab/dotnet)
+# [C# or .NET](#tab/dotnet)
```
Comment (heath-hamilton) [inline msteams-platform/bots/how-to/update-and-delete-bot-messages.md:131]:
```suggestion
# [C#](#tab/dotnet)
```

File: msteams-platform/bots/how-to/update-and-delete-bot-messages.md [hunk 6]
```diff
@@ -115,9 +139,9 @@ foreach (var activityId in _list)
 }
 ```
 
-# [TypeScript/Node.js](#tab/typescript)
+# [TypeScript or Node.js](#tab/typescript)
```
Comment (heath-hamilton) [inline msteams-platform/bots/how-to/update-and-delete-bot-messages.md:142]:
```suggestion
# [TypeScript](#tab/typescript)
```

**Review by heath-hamilton — CHANGES_REQUESTED** [review]
@v-rodriguezd Please see my suggested changes on how we label code examples. Thanks in advance!

File: msteams-platform/bots/calls-and-meetings/call-notifications.md [hunk 7]
```diff
@@ -1,42 +1,46 @@
...
 
 ```http
 HTTP/1.1 302 Found
 Location: your-new-location
 ```
 
-Enable your bot to answer an incoming call using the [answer](https://developer.microsoft.com/graph/docs/api-reference/beta/api/call_answer) API. You can specify the `callbackUri` to handle this particular call. This is useful for _stateful_ instances where your call is handled by a particular partition and you want to embed this information in the `callbackUri` for routing to the right instance.
+Enable your bot to answer an incoming call using the [answer](https://developer.microsoft.com/graph/docs/api-reference/beta/api/call_answer) API. You can specify the `callbackUri` to handle this particular call. This is useful for stateful instances where your call is handled by a particular partition and you want to embed this information in the `callbackUri` for routing to the right instance.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/call-notifications.md:35]: particular partition,

File: msteams-platform/bots/calls-and-meetings/call-notifications.md [hunk 8]
```diff
@@ -1,42 +1,46 @@
...
+
+The next section provides details on authenticating the callback by inspecting the token posted to your webhook.
 
-## Authenticating the callback
+## Authenticate the callback
 
-Your bot should inspect the token posted to your webhook to validate the request. Whenever the API posts to the webhook, the HTTP POST message contains an OAuth token in the Authorization header as a Bearer token, with audience as your application's App ID.
+Your bot must inspect the token posted to your webhook to validate the request. Every time the API posts to the webhook, the HTTP POST message contains an OAuth token in the Authorization header as a bearer token, with audience as your application's App ID.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/call-notifications.md:41]: with an/the audience 
possible?

File: msteams-platform/bots/calls-and-meetings/call-notifications.md [hunk 9]
```diff
@@ -68,10 +72,17 @@ The OAuth token will have values like the following, and will be signed by Skype
 }
 ```
 
+The OpenID configuration published at <https://api.aps.skype.com/v1/.well-known/OpenIdConfiguration> can be used to verify the token. Each of the OAuth token values are used as follows:
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/call-notifications.md:75]: Each of the OAuth token values is used as follows
Each of the OAuth token value is used as follows
Each singular

**Review by v-pmuthyala — CHANGES_REQUESTED** [review]
2 'new' pages are available for your review.

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 10]
```diff
@@ -0,0 +1,402 @@
...
+    "locale": "en-US"
+}
+```
+
+---
+
+> [!NOTE]
+> Message splitting occurs when a text message and an attachment are sent in the same activity payload. This activity is split into separate activities by Microsoft Teams, one with just a text message and the other with an attachment. As the activity is split, you do not receive the message ID in response, which is used to [update or delete](~/bots/how-to/update-and-delete-bot-messages.md) the message proactively. It is recommended to send separate activities instead of depending on message splitting.
+
+Messages sent between users and bots include internal channel data within the message. This data allows the bot to properly communicate on that channel. The Bot Builder SDK allows you to modify the message structure.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:202]: communicate properly

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 11]
```diff
@@ -0,0 +1,402 @@
...
+
+## Teams channel data
+
+The `channelData` object contains Teams-specific information and is a definitive source for team and channel IDs. You must cache and use these IDs as keys for local storage. The `TeamsActivityHandler` in the SDK, typically pulls out important information from the `channelData` object to make it easily accessible. However, you can always access the original data from the `turnContext` object.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:206]: in the SDK, typically - remove comma.

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 12]
```diff
@@ -0,0 +1,402 @@
...
+
+The `channelData` object is not included in messages in personal conversations, as these take place outside of a channel.
+
+A typical `channelData` object in an activity sent to your bot contains the following information:
+
+* `eventType`: Teams event type passed only in cases of [channel modification events](~/bots/how-to/conversations/subscribe-to-conversation-events.md).
+* `tenant.id`: Azure Active Directory tenant ID passed in all contexts.
+* `team`: Passed only in channel contexts not in personal chat.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:214]: contexts,

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 13]
```diff
@@ -0,0 +1,402 @@
...
+  * `id`: GUID for the channel.
+  * `name`: Name of the team passed only in cases of [team rename events](~/bots/how-to/conversations/subscribe-to-conversation-events.md).
+* `channel`: Passed only in channel contexts when the bot is mentioned or for events in channels in teams where the bot has been added.
+  * `id`: GUID for the channel.
+  * `name`: Channel name passed only in cases of [channel modification events](~/bots/how-to/conversations/subscribe-to-conversation-events.md).
+* `channelData.teamsTeamId`: Deprecated. This property is only included for backwards compatibility.
+* `channelData.teamsChannelId`: Deprecated. This property is only included for backwards compatibility.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:221]: backward

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 14]
```diff
@@ -0,0 +1,402 @@
...
+Messages received from or sent to your bot can include different types of message content.
+
+## Message content
+
+Your bot can send rich text, pictures, and cards. Users can send rich text and pictures to your bot.
+
+| Format    | From user to bot | From bot to user | Notes                                                                                   |
+|-----------|------------------|------------------|-----------------------------------------------------------------------------------------|
+| Rich text | ✔                | ✔                |                                                                                         |
+| Pictures  | ✔                | ✔                | Maximum 1024×1024 and 1 MB in PNG, JPEG, or GIF format; animated GIF are not supported.  |
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:252]: format; ';' ?

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 15]
```diff
@@ -0,0 +1,402 @@
...
+Pictures are sent by adding attachments to a message. For more information on attachments, see [Bot Framework documentation](/azure/bot-service/dotnet/bot-builder-dotnet-add-media-attachments).
+
+Pictures can be at most 1024×1024 and 1 MB in PNG, JPEG, or GIF format. Animated GIF is not supported.
+
+Specify the height and width of each image by using XML. In markdown, the image size defaults to 256×256. For example:
+
+* Use - `<img src="http://aka.ms/Fo983c" alt="Duck on a rock" height="150" width="223"></img>`
+* Don't use - `![Duck on a rock](http://aka.ms/Fo983c)`
+
+A conversational bot can include adaptive cards that simplify business workflows. Adaptive cards offer rich customizable text, speech, images, buttons and input fields.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:348]: buttons,

File: msteams-platform/bots/how-to/conversations/conversation-messages.md [hunk 16]
```diff
@@ -0,0 +1,402 @@
...
+
+## Adaptive cards
+
+Adaptive cards can be authored in a bot and shown in multiple apps such as Teams, your website and so on. For more information, see [adaptive cards](~/task-modules-and-cards/cards/cards-reference.md#adaptive-card).
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/conversation-messages.md:352]: website, and

**Review by v-pmuthyala — CHANGES_REQUESTED** [review]
3 pages

File: msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md [hunk 17]
```diff
@@ -2,43 +2,47 @@
...
+ms.topic: conceptual
 ms.author: anclear
 ---
-# Channel and group chat conversations with a Microsoft Teams bot
+# Channel and group chat conversations with a bot
 
 [!INCLUDE [pre-release-label](~/includes/v4-to-v3-pointer-bots.md)]
 
-By adding the `teams` or `groupchat` scope to your bot, it can be available to be installed in a team or group chat. This allows all members of the conversation to interact with your bot. Once installed, it will also have access to metadata about the conversation like the list of conversation members, and when installed in a team details about that team and the full list of channels.
+To install the Microsoft Teams bot in a team or group chat, add the `teams` or `groupchat` scope to your bot. This allows all members of the conversation to interact with your bot. After the bot is installed, it has access to metadata about the conversation, such as the list of conversation members. Also, when it is installed in a team, the bot has access to details about that team and the full list of channels.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md:12]: groupchat - from MSTP

File: msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md [hunk 18]
```diff
@@ -2,43 +2,47 @@
...
 
-When your bot is installed in a team, it can sometimes be necessary to create a new conversation thread rather than replying to an existing one. This is a form of [proactive messaging](~/bots/how-to/conversations/send-proactive-messages.md).
+## Create new conversation threads
 
-## Working with mentions
+When your bot is installed in a team, you must create a new conversation thread rather than reply to an existing one. At times it is difficult to differentiate between two conversations. If the conversation is threaded, it is easier to organize and manage different conversations in channels. This is a form of [proactive messaging](~/bots/how-to/conversations/send-proactive-messages.md).
 
-Every message to your bot from a group or channel will contain an @mention with its own name in the message text, so you'll need to ensure your message parsing handles that. Your bot can also retrieve other users mentioned in a message, and add mentions to any messages it sends.
+Next you can retrieve mentions using the `entities` object and add mentions to your messages using the `Mention` object.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md:29]: Next,

File: msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md [hunk 19]
```diff
@@ -2,43 +2,47 @@
...
 
-### Stripping mentions from message text
+## Work with mentions
 
-You may find it necessary to strip out the @mentions from the text of the message your bot receives.
+Every message to your bot from a group or channel contains an @mention with its own name in the message text. Ensure that your message parsing handles @mention. Your bot can also retrieve other users mentioned in a message and add mentions to any messages it sends.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md:33]: can we remove 'own'

File: msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md [hunk 20]
```diff
@@ -2,43 +2,47 @@
...
 
-### Retrieving mentions
+You must also strip out the @mentions from the text of the message your bot receives.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/how-to/conversations/channel-and-group-conversations.md:35]: content - instead of text

**Review by v-pmuthyala — CHANGES_REQUESTED** [review]
3 pages

File: msteams-platform/bots/how-to/create-a-bot-commands-menu.md [hunk 21]
```diff
@@ -2,57 +2,61 @@
...
+    ![App Studio commands menu Add button](./conversations/media/AppStudio-CommandMenu-Add.png)
 
-5. The **New Command** screen appears. Enter the **Command text** that you want to have appear as the menu command, and the **Help text** that you want to have appear directly under the command text in the menu. This should be a brief explanation of the purpose of the command.
+5. Enter the **Command text** that must appear as the command menu for your bot.
 
-6. Next, select the scope(s) where you want this command menu to appear, then select the **Save** button.
+6. Enter the **Help text** that must appear under the command text in the menu. **Help text** must be a brief explanation of the purpose of the command.
 
-    ![App Studio Command Menu Add button](./conversations/media/AppStudio-NewCommandMenu.png)
+7. Select the **Scope** check boxes to select where this command menu must appear and select **Save**.
```
[the remark this answers is not in the document]
Comment (v-rodriguezd) [reply msteams-platform/bots/how-to/create-a-bot-commands-menu.md:45]: Leaving as is as show up doesnt sound right

File: msteams-platform/bots/calls-and-meetings/real-time-media-concepts.md [hunk 22]
```diff
@@ -6,50 +6,59 @@ keywords: audio stream video stream audio/video calling meeting real-time media
...
+When a real-time media bot answers an incoming call or joins a Microsoft Teams meeting, it must declare what modalities it must support. For each supported modality, the bot declares whether it can send and receive media, receive only, or send only. For example, a bot designed to handle 1:1 Teams calls, requires to both send and receive audio, but only send video as it does not require to receive the video of the caller. The set of audio and video modalities established between the bot and the Teams caller or meeting is called the media session.
+
+Two types of video modalities are supported, main video and video-based screen sharing. Main video is used to transport the video from a user's webcam. Video-based screen sharing allows a user to share his or her screen as a video stream. The platform allows a bot to send and receive both video types.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/real-time-media-concepts.md:20]: The main video is

File: msteams-platform/bots/calls-and-meetings/real-time-media-concepts.md [hunk 23]
```diff
@@ -6,50 +6,59 @@ keywords: audio stream video stream audio/video calling meeting real-time media
...
 
 ## Video subscription
 
-In a 1:1 call, the bot will automatically receive the video of the caller if the bot is enabled to receive video. In a Teams meeting, the bot must indicate to the platform which participants it wants to see. A **video subscription** is a request by the bot to receive a participant’s main video or screen-sharing content. As the participants in the meeting conduct their conversation, the bot may modify its desired video subscriptions based on updates of the dominant speaker set or notifications indicating which participant is currently screen sharing.
+In a 1:1 call, the bot automatically receives the video of the caller if the bot is enabled to receive video. In a Teams meeting, the bot must indicate to the platform which participants it wants to see. A video subscription is a request by the bot to receive a participant’s main video or screen-sharing content. As the participants in the meeting conduct their conversation, the bot modifies its desired video subscriptions based on updates of the dominant speaker set or notifications indicating which participant is currently screen sharing.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/real-time-media-concepts.md:52]: enabled to receive the video.

File: msteams-platform/bots/calls-and-meetings/registering-calling-bot.md [hunk 24]
```diff
@@ -1,75 +1,90 @@
...
-  ![Configure Microsoft Teams channel information](~/assets/images/calls-and-meetings/configure-msteams-channel.png)
+The next section provides a list of application permissions supported for calls and online meetings.
 
-## Add Microsoft Graph permissions
+## Add Graph permissions
 
-Microsoft Graph exposes granular permissions controlling the access that apps have to resources. As a developer, you decide which permissions for Microsoft Graph your app requests.  The Microsoft Graph Calling APIs support _Application permissions_, which are used by apps that run without a signed-in user present.  A tenant administrator must grant consent to application permissions. Below is a list of those permissions:
+Graph provides granular permissions to control the access that apps have to resources. You decide which permissions for Graph your app requests. The Graph Calling APIs support application permissions, which are used by apps that run without a signed-in user present. A tenant administrator must grant consent to application permissions.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/registering-calling-bot.md:45]: The graph provides

File: msteams-platform/bots/calls-and-meetings/registering-calling-bot.md [hunk 25]
```diff
@@ -1,75 +1,90 @@
...
 
 > [!IMPORTANT]
-> You **cannot** use the Media Access API to record or otherwise persist media content from calls or meetings that your application accesses, or data derived from that media content ("record" or "recording"), without first calling the [`updateRecordingStatus` API](/graph/api/call-updaterecordingstatus) to indicate that recording has begun, and receiving a success reply from that API. If your application begins recording any meeting/call, it must end the recording prior to calling the `updateRecordingStatus` API to indicate that the recording has ended.
+> You cannot use the Media Access API to record or otherwise persist media content from calls or meetings that your application accesses or data derived from that media content record or recording. You must first call the [`updateRecordingStatus` API](/graph/api/call-updaterecordingstatus) to indicate that recording has begun, and receive a success reply from that API. If your application begins recording any meeting or call, it must end the recording prior to calling the `updateRecordingStatus` API to indicate that the recording has ended.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/registering-calling-bot.md:60]: it must end the recording before calling the

File: msteams-platform/bots/calls-and-meetings/registering-calling-bot.md [hunk 26]
```diff
@@ -1,75 +1,90 @@
...
 
 > [!NOTE]
-> Constructing the Tenant Admin Consent URL requires a configured Redirect URI/Reply URL in the [App Registration Portal](https://apps.dev.microsoft.com/). To add reply URLs for your bot, access your bot registration, choose Advanced Options -> Edit Application Manifest.  Add your Redirect URL to the `replyUrls` collection.
+> To construct the tenant Admin consent URL, a configured redirect URI or reply URL in the [app registration portal](https://apps.dev.microsoft.com/) is required. To add reply URLs for your bot, access your bot registration, choose **Advanced Options** -> **Edit Application Manifest**.  Add your redirect URL to the `replyUrls` collection.
 
 > [!IMPORTANT]
-> Anytime you make a change to your application's permissions, you must also repeat the Admin Consent process. Changes made in the app registration portal will not be reflected until consent has been reapplied by the tenant's administrator.
+> Anytime you make a change to your application's permissions, you must also repeat the Admin consent process. Changes made in the app registration portal are not reflected until consent has been reapplied by the tenant's administrator.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/registering-calling-bot.md:85]: until the consent

File: msteams-platform/bots/calls-and-meetings/call-notifications.md [hunk 27]
```diff
@@ -1,42 +1,47 @@
...
+The next section provides details on authenticating the callback by inspecting the token posted to your webhook.
 
-## Authenticating the callback
+## Authenticate the callback
 
-Your bot should inspect the token posted to your webhook to validate the request. Whenever the API posts to the webhook, the HTTP POST message contains an OAuth token in the Authorization header as a Bearer token, with audience as your application's App ID.
+Your bot must inspect the token posted to your webhook to validate the request. Every time the API posts to the webhook, the HTTP POST message contains an OAuth token in the Authorization header as a bearer token, with the audience as your application's App ID.
 
-Your application should validate this token before accepting the callback request.
+Your application must validate this token before accepting the callback request.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/call-notifications.md:44]: : required?

File: msteams-platform/bots/calls-and-meetings/requirements-considerations-application-hosted-media-bots.md [hunk 28]
```diff
@@ -8,46 +8,65 @@ ms.date: 11/16/2018
 
 # Requirements and considerations for application-hosted media bots
 
-Not all guidance for developing messaging and Interactive Voice Response (IVR) bots applies equally to building application-hosted media bots. This article describes some of the important requirements and considerations for developing and running an application-hosted media bot.
+An application-hosted media bot requires the [`Microsoft.Graph.Communications.Calls.Media` .NET library](https://www.nuget.org/packages/Microsoft.Graph.Communications.Calls.Media/) to access the audio and video media streams, and the bot must be deployed on a Windows Server machine or Windows Server guest Operating System (OS) in Azure.
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/requirements-considerations-application-hosted-media-bots.md:11]: media streams and the bot

File: msteams-platform/bots/calls-and-meetings/requirements-considerations-application-hosted-media-bots.md [hunk 29]
```diff
@@ -8,46 +8,65 @@ ms.date: 11/16/2018
...
 
-## Samples and Additional Resources
+- The bots require more compute and network bandwidth capacity than messaging bots and incur significantly higher operational costs. A real-time media bot developer must carefully measure the bot's scalability, and ensure the bot does not accept more simultaneous calls than it can manage. A video-enabled bot can sustain only one or two concurrent media sessions per CPU core if using the raw RGB24 or NV12 video formats.
+- The Real-time Media Platform does not currently take advantage of any Graphics Processing Units (GPU) available on the VM to off-load H.264 video encoding or decoding. Instead, video encode and decode are done in software on the CPU. If a GPU is available, the bot takes advantage of it for its own graphics rendering, for example, if the bot is using a 3D graphics engine.
+- The VM instance hosting the real-time media bot must have at least 2 CPU cores. For Azure, a Dv2-series virtual machine is recommended. For other Azure VM types, a system with 4 virtual CPUs (vCPU) is the minimum size required. For more information about Azure VM types, se [Azure documentation](/azure/virtual-machines/windows/sizes-general).
```
Comment (v-pmuthyala) [inline msteams-platform/bots/calls-and-meetings/requirements-considerations-application-hosted-media-bots.md:60]: information about Azure VM types, see [Azure documentation

File: msteams-platform/resources/team-chat-member-api-changes.md [hunk 30]
```diff
@@ -6,21 +6,23 @@ keywords: bot framework apis team members roster
...
+
+# Teams bot API changes to fetch team or chat members
 
 >[!NOTE]
-> We've started with the deprecation process for `TeamsInfo.getMembers` and `TeamsInfo.GetMembersAsync` APIs. Initially, they will be heavily throttled to 5 requests per minute and return a maximum of 10K members per team. This will result in the full roster not being returned as team size increases. 
-> 
-> **You must update to version 4.10 or higher of the Bot Framework SDK** and switch to the paginated API endpoints, or the `TeamsInfo.GetMemberAsync` single user API. This also applies to your bot even if you're not directly using these APIs, as older SDKs call these APIs during [membersAdded](../bots/how-to/conversations/subscribe-to-conversation-events.md#team-members-added) events. To view the list of upcoming changes, see [API changes](team-chat-member-api-changes.md#api-changes). 
+> The deprecation process for `TeamsInfo.getMembers` and `TeamsInfo.GetMembersAsync` APIs has started. Initially, they are heavily throttled to 5 requests per minute and return a maximum of 10K members per team. This results in the full roster not being returned as team size increases.
```
Comment (v-pmuthyala) [inline msteams-platform/resources/team-chat-member-api-changes.md:13]: APIs have started.

File: msteams-platform/resources/team-chat-member-api-changes.md [hunk 31]
```diff
@@ -50,16 +52,22 @@ Here’s a sample JSON representation of what’s returned by these APIs today.
...
+
+    > [!NOTE]
+    > In either v3 or v4, the best action is to upgrade to the latest point release that is 3.30.2 or 4.8 respectively.
+
+* A new API is created [`TeamsInfo.GetMemberAsync`](https://docs.microsoft.com/microsoftteams/platform/bots/how-to/get-teams-context?tabs=dotnet#get-single-member-details) for retrieving the profile information for a single user. It takes the ID of the team or chat and a [UPN](https://docs.microsoft.com/windows/win32/ad/naming-properties#userprincipalname) that is `userPrincipalName`, Azure Active Directory Object ID `objectId`, or the Teams user ID `id` as parameters and returns the profile information for that user.
+
+    > [!NOTE]
+    > `objectId` is changed to `aadObjectId` to match what it is called in the `Activity` object of a Bot Framework message. The new API is available with version 4.8 of the Bot Framework SDK. It is available in the Teams SDK extension Bot Framework 3.x as well. Meanwhile you can use the [REST](https://docs.microsoft.com/microsoftteams/platform/bots/how-to/get-teams-context?tabs=json#get-single-member-details) endpoint.
```
Comment (v-pmuthyala) [inline msteams-platform/resources/team-chat-member-api-changes.md:66]: Meanwhile,

**Review by v-pmuthyala — CHANGES_REQUESTED** [review]
completed

File: msteams-platform/bots/how-to/bots-filesv4.md [hunk 32]
```diff
@@ -74,9 +78,9 @@ The following table describes the content properties of the attachment:
 
 As a best practice, acknowledge the file upload by sending a message back to the user.
 
-### Uploading files to personal chat
+### Upload files to personal chat
 
-The following steps are required to upload a file to a user:
+**To upload a file to a user**
```
[the remark this answers is not in the document]
Comment (v-rodriguezd) [reply msteams-platform/bots/how-to/bots-filesv4.md:84]: We dont give colon in this case

File: msteams-platform/bots/calls-and-meetings/calls-meetings-bots-overview.md [hunk 33]
```diff
@@ -6,37 +6,82 @@ keywords: calling calls audio video IVR voice online meetings
...
+
+Now you must understand some core concepts, terminology, and conventions.
+
+## Terminologies
+
+The following core concepts, terminology, and conventions guide you through the use of calls and online meetings bots:
```
[the remark this answers is not in the document]
Comment (v-rodriguezd) [reply msteams-platform/bots/calls-and-meetings/calls-meetings-bots-overview.md:29]: yes it is introduction sentence for the list

**Review by v-pmuthyala — DISMISSED** [review]
changes are reviewed.

**Review by v-pmuthyala — DISMISSED** [review]
reviewed.

**Comment by KirtiPereira** [comment]
@Wajeed-msft, @ojasvichoudhary, and @Trinetra-MSFT
 
A gentle reminder for tech review/sign-off.

Please provide your comments as we will be merging this PR on Thursday, April 8th. If we do not receive any comments, we will still merge the PRs.
In case of any gaps, please raise a doc-bug.  

Thank you.

**Review by KirtiPereira — APPROVED** [review]
Approved

**Review by surbhigupta12 — APPROVED** [review]
Approved the PR

**Review by heath-hamilton — DISMISSED** [review]
Approved for publication, @surbhigupta12 and @KirtiPereira.

**Review by surbhigupta12 — APPROVED** [review]
Approving the PR

**Review by KirtiPereira — APPROVED** [review]
Approved.

**Review by surbhigupta12 — APPROVED** [review]
Approving PR

**Pull request merged**